platform/src/main/kotlin/uk/gov/gdx/datashare/services/AcquirersService.kt

```diff
@@ -73,8 +76,15 @@ class AcquirersService(
 
   private fun addAcquirerSubscriptionEnrichmentFields(
     acquirerSubscriptionId: UUID,
+    eventType: EventType,
     enrichmentFields: List<EnrichmentField>,
   ): List<AcquirerSubscriptionEnrichmentField> {
+    val eventTypeEnrichmentFields = EventTypeEnrichmentFieldsRelationship[eventType]
+    enrichmentFields.forEach {
+      if (!eventTypeEnrichmentFields!!.contains(it)) {
+        throw EnrichmentFieldsNotValidForEventType("Invalid enrichment field(s) for the given event type")
```

Comment:
could you add in which enrichment field and which event type into the error message. That would involve add the failing enrichment fields to a list and then throwing if that list has any in it.